app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserMenuSignIn.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import android.content.Context
+import android.view.View
+import android.widget.TextView
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.core.content.ContextCompat
+import mozilla.components.browser.menu.BrowserMenu
+import mozilla.components.browser.menu.item.BrowserMenuImageText
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+
+internal const val NO_ID = -1
+
+internal fun TextView.setColorResource(@ColorRes textColorResource: Int) {
+    if (textColorResource != NO_ID) {
+        setTextColor(ContextCompat.getColor(context, textColorResource))
+    }
+}
```

Comment:
Looking at the code, it didn't appear we would ever hit the `NO_ID` case since we are always passing in the primary color. Do we need this?

Reply:
I copied it from the parent class, to follow the same parameter validation rules. I am not sure we need it, since we are the only users of the class. Removing.